## TEAM 24

In [2]:
import csv
import json
from statsmodels.stats.inter_rater import fleiss_kappa
import numpy as np
from sklearn.metrics import cohen_kappa_score

## POS : Cohen Kappa

In [3]:
POS_LABELS = [ "NOUN", "PROPN", "VERB", "ADJ", "ADV", "ADP", "PRON", "DET", "CONJ", "PART", "PRON_WH", "PART_NEG", "NUM", "X"]

In [4]:
# function to load data and return an array of label value for each token
def POS_data_arr(csv_file,i):
    data = []
    with open(csv_file) as csvfile:
        reader = csv.DictReader(csvfile)
        for row in reader:
            col = json.loads(row["label"])
            for labels in col:
                labels = labels["labels"]
                for label in labels:
                    # find the index of label from POS_LABELS and append it to the data list
                    data.append(POS_LABELS.index(label))
    return np.array(data)

In [10]:
# get label arrays
POS_labels0 = POS_data_arr("POS_annotation_srajan.csv",0)
POS_labels1 = POS_data_arr("POS_annotation_rahul.csv",1)

# calculate cohen_kappa
cohen_kappa = cohen_kappa_score(POS_labels0,POS_labels1)
print(f"Cohen kappa score: {cohen_kappa}")

Cohen kappa score: 0.684578779


The Cohen kappa score is near to 1 implying there was very high agreement between the annotators

## CV : Fleiss Kappa

In [6]:
# function to load data and return an array of label value for each token
def CV_data_matrix(csv_file,i):
    data = []
    with open(csv_file) as csvfile:
        reader = csv.DictReader(csvfile)
        for row in reader:
            # first col indicates 'Trucks' and second col indicates 'No Trucks'
            data.append([0,1] if row["choice"]=="No Trucks" else [1,0])
    return np.array(data)

In [7]:
# load and get label matrices
CV_labels0 = CV_data_matrix("CV_annotation_srajan.csv",0)
CV_labels1 = CV_data_matrix("CV_annotation_rahul.csv",1)
CV_labels2 = CV_data_matrix("CV_annotation_other.csv",2)

CV_labels = CV_labels0 + CV_labels1 + CV_labels2

value = fleiss_kappa(CV_labels)
print(f"Fleiss kappa score: {value}")

Fleiss kappa score: 0.22077922077922085


The Fleiss kappa score is about 0.22 implying there was fair degree of agreement between the annotators